pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -129,7 +129,10 @@ func NewController(
 	})
 
 	c.Logger.Info("Setting up ConfigMap receivers")
-	c.configStore = config.NewStore(c.Logger.Named("config-store"))
+	resyncRoutesOnConfigDomainChange := configmap.TypeFilter(&config.Domain{})(func(string, interface{}) {
+		impl.GlobalResync(routeInformer.Informer())
+	})
+	c.configStore = config.NewStore(c.Logger.Named("config-store"), resyncRoutesOnConfigDomainChange)
```

Comment:
I find this a little hard to read as-is, but we can iterate as we wake up smarter.